app/com/github/pedrovgs/dotto/interpreter/led/ledController.scala

```diff
@@ -0,0 +1,74 @@
+package com.github.pedrovgs.dotto.interpreter.led
+
+import com.github.pedrovgs.dotto.core.algebra.dotto.{LedInteractions, MorseSentence}
+import com.github.pedrovgs.dotto.core.types._
+import com.pi4j.io.gpio._
+import play.api.Logger
+
+import scala.concurrent.duration._
+
+object ledController {
+
+  private val DotDuration = 0.1 seconds
+  private val DashDuration = DotDuration * 3
+  private val SpaceDuration = DashDuration * 3
+  private val SpaceBetweenSymbolsDuration = DotDuration * 2
+
+  private val SpaceBetweenSymbolsInteraction = LedInteraction(Low, SpaceBetweenSymbolsDuration)
+
+  /**
+    * Transforms a sequence of sequences of MorseSymbol instances into a sequence of led interactions. Every word
+    * inside the sentence represented with a sequence of MorseSymbols will be separated to the rest of the words using a
+    * space led interaction.
+    */
+  def toLedInteractions(morseSentence: MorseSentence): LedInteractions = {
+    morseSentence.flatten.map {
+      case Dot => LedInteraction(High, DotDuration)
+      case Dash => LedInteraction(High, DashDuration)
+      case Space => LedInteraction(Low, SpaceDuration)
+    }.flatMap(Seq(_, SpaceBetweenSymbolsInteraction))
+  }
+
+  /**
+    * Based on a led interactions sequence manipulates the led configured in the GPIO_26 pin using PI4J library.
+    */
+  def showMorseSentenceInLed(ledInteractions: LedInteractions) = {
+    Logger.debug("Let's show this led interactions using the GPIO API: " + ledInteractions)
+    val gpio = GpioFactory.getInstance()
+    val pin = initGPIO26Pin(gpio)
+    showLedInteractionsInLed(ledInteractions, pin)
```

Comment:
These are effects with discarding values they should be wrapperd in your ADT by providing effect commands for them or lifted to an effect capturing monad that can deal with it's impure execution by placing results in a disjunction type container. See monix Task or any of the other Task implementations.